File: step06_appsync_frontend_amplify/backend-cdk/functions/deleteTodo.ts
const AWS = require('aws-sdk');
const docClient = new AWS.DynamoDB.DocumentClient();

async function deleteTodo(todoId: string) {
    const params = {
        TableName: process.env.TODOS_TABLE,
        Key: {
            id: todoId
        }
    }
    try {
        await docClient.delete(params).promise()
        return todoId
    } catch (err) {
        console.log('DynamoDB error: ', err)
        return null
    }
}

export default deleteTodo;

File: step06_appsync_frontend_amplify/backend-cdk/python/functions/deleteTodo.py
import boto3
import os
import logging
from botocore.exceptions import ClientError
dynamodb = boto3.resource('dynamodb')
tableName = os.environ['TODOS_TABLE']
table = dynamodb.Table(tableName)


def deleteItem(todoId):
    try:
        res = table.delete_item(
            Key={
                'id': todoId
            }
        )

        return todoId
    except ClientError as e:
        logging.error(e)
